components/webfield/ConsoleTabs.js

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'react'
+import { useRouter } from 'next/router'
+import { Tab, TabList, TabPanel, TabPanels, Tabs } from '../Tabs'
+
+const ConsoleTabs = ({ defaultActiveTabId, tabs = [], updateActiveTabId }) => {
+  const validTabIds = tabs.flatMap((tab) => (tab.visible ? tab.id : []))
+  const [activeTabId, setActiveTabId] = useState(
+    decodeURIComponent(window.location.hash.substring(1)) ||
+      defaultActiveTabId ||
+      validTabIds[0]
+  )
+  const router = useRouter()
+
+  useEffect(() => {
+    if (!validTabIds.includes(activeTabId)) {
+      setActiveTabId(defaultActiveTabId)
+      updateActiveTabId?.(`#${defaultActiveTabId}`)
+      return
+    }
```

Comment:
Do we need to call `updateActiveTabId` here? I think when `setActiveTabId` is called it will trigger useEffect again with a valid (default) tab, so the parent's state would already get updated in line 20 with the default. What do you think?